packages/api/src/command/webhook/fitbit.ts

```diff
@@ -0,0 +1,176 @@
+      return { cxId: connectedUser.cxId, userId: connectedUser.id, typedData: userFitbitData };
+    })
+  );
+
+  const reducedData: Entry[] = [];
+
+  connectedUsersAndData.forEach(entry => {
+    const existingUserIndex = reducedData.findIndex(
+      item => item.cxId === entry.cxId && item.userId === entry.userId
+    );
+
+    if (existingUserIndex >= 0) {
+      const entryKeys = Object.keys(entry.typedData) as DataType[];
+
+      reducedData[existingUserIndex] = {
+        ...reducedData[existingUserIndex],
+        typedData: {
+          ...reducedData[existingUserIndex].typedData,
+          ...(entryKeys.includes("activity")
+            ? {
+                activity: union(
+                  reducedData[existingUserIndex].typedData.activity,
+                  entry.typedData.activity
+                ),
+              }
+            : undefined),
+          ...(entryKeys.includes("nutrition")
+            ? {
+                nutrition: union(
+                  reducedData[existingUserIndex].typedData.nutrition,
+                  entry.typedData.nutrition
+                ),
+              }
+            : undefined),
+          ...(entryKeys.includes("body")
+            ? { body: union(reducedData[existingUserIndex].typedData.body, entry.typedData.body) }
+            : undefined),
+          ...(entryKeys.includes("sleep")
+            ? {
+                sleep: union(reducedData[existingUserIndex].typedData.sleep, entry.typedData.sleep),
+              }
+            : undefined),
+        },
+      };
+    } else {
+      reducedData.push(entry);
+    }
+  });
+
+  const dataByCustomer = groupBy(reducedData, v => v.cxId);
+
+  await Promise.allSettled(
+    Object.keys(dataByCustomer).map(async cxId => {
+      try {
+        const dataAndUserList = dataByCustomer[cxId].map(v => ({
+          userId: v.userId,
+          typedData: v.typedData,
+        }));
+
+        const transformedData = dataAndUserList.map(userData => {
+          const { userId, typedData } = userData;
+          return { userId, ...typedData };
+        });
+
+        const payload: WebhookDataPayloadWithoutMessageId = { users: transformedData };
+
+        const settings = await getSettingsOrFail({ id: cxId });
+
+        analytics({
+          distinctId: cxId,
+          event: EventTypes.query,
+          properties: {
+            method: "POST",
+            url: "/webhook/fitbit",
+            apiType: ApiTypes.devices,
+          },
+        });
+        const webhookRequest = await createWebhookRequest({ cxId, payload });
+        // send it to the customer and update the request status
+        await processRequest(webhookRequest, settings);
+
+        reportDevicesUsage(
+          cxId,
+          dataAndUserList.map(du => du.userId)
+        );
+      } catch (err) {
+        log(`Error on processData: ${err}`);
+        capture.error(err, {
+          extra: { context: `webhook.processData.fitbit`, err },
+        });
+      }
```

Comment:
I'd move this to a separated function, so `processData` is leaner and easier to test (could also move the block before so `processData` becomes "getDataByCustomer" and "sendDataToCustomer").